javascript/javascript-import-and-require.md

```diff
@@ -4,63 +4,104 @@
 
 Javascript helps us out with this by having ‘imports’ and ‘require’. This is how you can write code in one file and share that code so it can be used by another file or files
 
-**What are differences between Import and Require?**
+**What are the differences between IMPORT and REQUIRE?**
 |Import|Require|
 |---|---|
 |As import remains in stage three and not<br> enforced by browsers natively, we’re unable to<br> run any performance take a look at.|Presently once you use import in your code,<br> your transpilers it back to require,<br> the commonJS modeling system. <br>Therefore for nowadays each is the same|
 |Though there aren’t any performance profit at the instant,<br> however, I’ll still counsel to use import over<br> require because it’s about to be native in JS|As a result of import is native therefore<br>require doesn’t perform higher as compare to import|
 |You will have dynamic loading wherever the loaded module<br> name is not predefined.Loading is synchronous.<br> Meaning if you have got multiple requires,<br> they’re loaded and processed one by one.|Import is asynchronous and may perform a touch higher.<br> Also, the require module system is not<br> customarily based mostly. |
 
-**when we can and cannot use import in our Web Application?**
+**when we can and cannot use IMPORT in our Web Application?**
 
-The import statement is used to import bindings which are exported by another module. Imported modules are in strict mode whether you declare them as such or not. The import statement cannot be used in embedded scripts unless such script has a type="module".
+The **import** statement is used to get bindings which are exported by another module and imported modules are in strict mode whether you declare them as such or not. There is one condition when **import** statement cannot be used in embedded scripts, till the time you delcare type="module" in script file.
 
-**why neither works in the browser unless you fake it with webpack or equivalent?**
+**why IMPORT and REQUIRE neither works in the browser unless you fake it with webpack or equivalent?**
 
-Webpack tries to handle all require calls during application working, even those not meant for browser usage which makes it almost impossible to have a module that works in both Node.js and it creates browser problem to work with it.
+In javascript two different moules cannot be used togather, If one cannot be imported. So, we use [Webpack](https://insights.untapt.com/webpack-import-require-and-you-3fd7f5ea93c0) which have one impressive feature which is called tree shaking. This technique is uses to reduce the size of your javascript bundle.

```

Comment:
Tree shaking is not the correct explanation of `why import and require doesn't work in the browser without webpack`. Please correct it.

Reply:
Can you suggest me, I am little new to javascript